lib/galaxy/util/properties.py

```diff
@@ -105,9 +92,40 @@ def load_app_properties(
             if config_key not in properties:
                 properties[config_key] = os.environ[key]
 
+    # update from kwds
+    if kwds:
+        properties.update(kwds)
+
     return properties
 
 
+def read_properties_from_file(config_file, config_section=None):
+    properties = {}
```

Comment:
Add ``__file__`` to the properties?

Reply:
It is already added: if file is ini, it will be added here:
https://github.com/galaxyproject/galaxy/pull/8462/files#diff-a26aaaf06d468fa5e49ccb0c905e9051R130
if it's yaml, then here:
https://github.com/galaxyproject/galaxy/pull/8462/files#diff-a26aaaf06d468fa5e49ccb0c905e9051R107
And if there's no file, then here: 
https://github.com/galaxyproject/galaxy/pull/8462/files#diff-a26aaaf06d468fa5e49ccb0c905e9051R82

Reply:
links in comment were wrong, corrected now.

Reply:
it's also tested in the unit tests: all default properties are added in both cases (ini and yaml)

Reply:
Ahh - I see it now. Cool!